src/main/java/jenkins/branch/BranchBuildStrategy.java

```diff
@@ -160,13 +161,46 @@ public boolean isAutomaticBuild(@NonNull SCMSource source,
      * {@link SCMHead} has been detected as created / modified.
      * @since 2.4.2
      */
+    @Deprecated
+    @SuppressWarnings("deprecation")
+    @Restricted(ProtectedExternally.class)
+    public boolean isAutomaticBuild(@NonNull SCMSource source,
+                                             @NonNull SCMHead head,
+                                             @NonNull SCMRevision currRevision,
+                                             @CheckForNull SCMRevision lastBuiltRevision,
+                                             @CheckForNull SCMRevision lastSeenRevision,
+                                             @NonNull TaskListener listener){
+        throw new UnsupportedOperationException("Modern implementation accessed using legacy API method");
```

Comment:
this seems like a breaking change?  if an Extension overrides this calss and updated to the modern way removing the over ride on this deprecated function - then any consumers that have not updated will not blow up at runtime?

If the extension can do something about it - then it would seem this method is just un-needed - or could be used to call the modern method using the 
`Util.isOverridden` trick.

Reply:
I actually followed the example of what the rest of the plugin extension did.  IE - Line [111](https://github.com/jenkinsci/branch-api-plugin/blob/master/src/main/java/jenkins/branch/BranchBuildStrategy.java#L111) was originally shipped with 2.0.17 while line [138](https://github.com/jenkinsci/branch-api-plugin/blob/master/src/main/java/jenkins/branch/BranchBuildStrategy.java#L138) was shipped with 2.1.3.  Basically everytime there is a new version, the previous version is set to `UnsupportedOperationException`.

So the way I understand it the SPI method finds out dynamically which method is overridden (original, V2, V3, etc) and then calls that specific override.  So the SPI method uses `Util.isOverridden` to determine which API method to call.